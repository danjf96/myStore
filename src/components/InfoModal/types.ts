import { ImageRequireSource, ModalProps } from "react-native";

export interface PROPSMODALINFO extends ModalProps {
    image?: ImageRequireSource
    title: string,
    text: string,
    textButton?: string,
    onPress: () => void
    textSecondButton?: string,
    onPressSecondButton?: ()=> void
}
