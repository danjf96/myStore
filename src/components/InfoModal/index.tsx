
import React from 'react'
import { Modal, Text, TouchableOpacity, View } from 'react-native'
import { Image } from 'react-native-animatable'
import Styles, { StylesFooter } from './styles'
import { PROPSMODALINFO } from './types'

const InfoModal = (props:PROPSMODALINFO) => {
    const { image, text, title, textButton = 'OK, ENTENDI', onPress, visible } = props
    return (
        <>
            <Modal 
                animationType="fade"
                transparent={true}
                visible={visible}
                {...props}
            >
                <View style={Styles.modal}>
                    <View style={Styles.containerBackground}></View>

                    <View style={Styles.containerTwo}>

                        <View style={Styles.container}>
                            
                            {image && <Image style={Styles.image} source={image} /> }

                            <Text style={Styles.title}>{title}</Text>

                            <Text style={Styles.text}>{text}</Text>

                            <View style={StylesFooter.container}>

                                <TouchableOpacity style={StylesFooter.style} onPress={onPress}>
                                    <Text style={StylesFooter.textStyle}>{textButton}</Text>
                                </TouchableOpacity>

                                {
                                props.textSecondButton != undefined && props.textSecondButton != '' && 
                                    <TouchableOpacity style={StylesFooter.styleSecond} onPress={props.onPressSecondButton}>
                                        <Text style={StylesFooter.textSecondStyle}>{props.textSecondButton}</Text>
                                    </TouchableOpacity>
                                }

                            </View>
                        

                        </View>
                    </View>
                </View>            
            </Modal>
        </>
    )
}


export default InfoModal