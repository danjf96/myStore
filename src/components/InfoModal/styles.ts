import { StyleSheet } from "react-native";
import Colors from "../../assets/Colors";

const Styles = StyleSheet.create({
    modal: { 
        flex: 1, 
        alignItems: 'center', 
        justifyContent: 'center'
    },

    containerBackground: {
        width:'100%',
        height:'100%',
        backgroundColor: Colors.backgroundModal,
        opacity: 0.8,
        flexDirection: 'column',
        justifyContent:'center',
        alignItems:'center',
        position: 'absolute'
    },

    container: {
        width: '100%',
        padding: 24,
        paddingTop: 32,
        backgroundColor: Colors.secondary,
        borderRadius: 10
    },

    image: {
        width: 48, 
        height: 48,
        resizeMode: 'contain',
        marginBottom: 32
    },

    title: {
        fontSize: 20,
        lineHeight: 19.6,
        color: Colors.title,
        marginBottom: 24,
        fontWeight: 'bold'
    },

    text: {
        fontSize: 16,
        lineHeight: 24,
        letterSpacing: 0.15,
        marginBottom: 24,
        color: Colors.text,
        textAlign: 'left'
    },

    containerTwo: {
        paddingHorizontal: 40, 
        width: '100%' 
    }

})

export const StylesFooter = StyleSheet.create({

    container: {
        width: '100%'
    },

    style: {
        backgroundColor: Colors.principal,
        height: 56,
        borderRadius: 10,
        width: '100%',
        padding: 10,
        alignItems: 'center',
        justifyContent: 'center' ,
        marginBottom: 15   
    },

    textStyle: {
        color: Colors.textSecondary,
        fontSize: 16,
        fontWeight: 'bold'
    },

    textSecondStyle: {
        color: Colors.principal,
        fontSize: 16,
        fontWeight: 'bold'
    },

    styleSecond: {
        backgroundColor: Colors.secondary,
        height: 56,
        borderRadius: 10,
        width: '100%',
        padding: 10,
        alignItems: 'center',
        justifyContent: 'center',
        borderColor: Colors.principal,
        borderWidth: 1.5
    }
})

export default Styles