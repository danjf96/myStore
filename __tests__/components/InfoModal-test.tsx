/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import InfoModal from '../../src/components/InfoModal';
 import { fireEvent, render } from 'react-native-testing-library';
 
describe('Render component', () => {
    it('renders correctly', async () => { 
        const screen = render( <InfoModal image={0} title={''} text={''} onPress={function (): void {
            throw new Error('Function not implemented.');
        } } />)
        expect(screen.toJSON()).toMatchSnapshot()
    });

    test('text and functions', async () => {
        const onPress = jest.fn();
        const onPressSecond = jest.fn();
        const screen = render( <InfoModal image={0} 
                title={'TITLE'} 
                text={'TEXT'} 
                textButton={'ONE'} 
                textSecondButton='TWO' 
                onPress={onPress} 
                onPressSecondButton={onPressSecond}
            />
        )
        const { getByText, getAllByText } = screen

        expect(getAllByText(/TITLE|TEXT|ONE|TWO/)).toHaveLength(4);

        fireEvent(getByText('ONE'), 'press');
        expect(onPress).toHaveBeenCalled();

        fireEvent(getByText('TWO'), 'press');
        expect(onPressSecond).toHaveBeenCalled();

    })
 
 })