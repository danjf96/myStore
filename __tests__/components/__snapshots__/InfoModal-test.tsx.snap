// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  image={0}
  onPress={[Function]}
  text=""
  title=""
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353535",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "center",
          "opacity": 0.8,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "paddingHorizontal": 40,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 10,
            "padding": 24,
            "paddingTop": 32,
            "width": "100%",
          }
        }
      >
        0
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
              "lineHeight": 19.6,
              "marginBottom": 24,
            }
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "letterSpacing": 0.15,
              "lineHeight": 24,
              "marginBottom": 24,
              "textAlign": "left",
            }
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#0086ff",
                "borderRadius": 10,
                "height": 56,
                "justifyContent": "center",
                "marginBottom": 15,
                "opacity": 1,
                "padding": 10,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              OK, ENTENDI
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
